src/Dashboard/EditDashboard.js

```diff
@@ -39,12 +30,31 @@ class EditDashboard extends Component {
         }
     }
 
+    getDashboardContent = () => {
+        const Content = () => {
+            return this.props.updateAccess ? (
+                <DashboardContent editMode={true} />
+            ) : (
+                <NoContentMessage text={i18n.t('No access')} />
+            );
+        };
+
+        const contentNotReady =
+            !this.props.dashboardsLoaded || this.props.selectedId === null;
+
+        return (
+            <div className="dashboard-wrapper">
+                {contentNotReady ? null : <Content />}
```

Comment:
Just to be nitpicky the Content component could be created only if it's needed. Like:

{contentNotReady ? null : this.getContent(this.props.updateAccess)}